spec/setting_spec.rb

```diff
@@ -2,74 +2,79 @@
     s.value.should be false
 
-    s = ActiveSetting::Setting.new(:data_type => :boolean, :raw_value => '1')
+    s = ActiveSetting::Setting.new(data_type: :boolean, raw_value: '1')
     s.value.should be true
 
-    s = ActiveSetting::Setting.new(:data_type => :boolean, :raw_value => 'true')
+    s = ActiveSetting::Setting.new(data_type: :boolean, raw_value: 'true')
     s.value.should be true
   end
 
   it 'should handle multi value types including subvalue types and handle spacing' do
-    s = ActiveSetting::Setting.new(:data_type => :csv, :subtype => :integer, :raw_value => '1,2,3')
-    s.value.should == [1, 2, 3]
+    s = ActiveSetting::Setting.new(data_type: :csv, subtype: :integer, raw_value: '1,2,3')
+    s.value.should eq [1, 2, 3]
 
-    s = ActiveSetting::Setting.new(:data_type => :csv, :subtype => :symbol, :raw_value => 'first, second')
-    s.value.should == [:first, :second]
+    s = ActiveSetting::Setting.new(data_type: :csv, subtype: :symbol, raw_value: 'first, second')
+    s.value.should eq [:first, :second]
 
-    s = ActiveSetting::Setting.new(:data_type => :hash, :raw_value => 'a:1 , b : 2')
-    s.value.should == {:a => '1', :b => '2'}
+    s = ActiveSetting::Setting.new(data_type: :hash, raw_value: 'a:1 , b : 2')
+    s.value.should eq(a: '1', b: '2')
   end
 end
 
 describe ActiveSetting::Setting, 'when having options' do
   it 'should format the options in the right type' do
-    s = ActiveSetting::Setting.new(:options => 'easy normal hard')
-    s.options.should == ['easy', 'normal', 'hard']
+    s = ActiveSetting::Setting.new(options: 'easy normal hard')
+    s.options.should eq %w(easy normal hard)
   end
 
   it 'should format the options by key,value when object options' do
-    first  = double(:id => '1', :name => 'First')
-    second = double(:id => '2', :name => 'Second')
+    first  = double(id: '1', name: 'First')
+    second = double(id: '2', name: 'Second')
     s = ActiveSetting::Setting.new
-    s.objects_from_collection([first, second], :name, :id).should == [['First', '1'], ['Second', '2']]
+    s.objects_from_collection([first, second], :name, :id).should eq [%w(First 1), %w(Second 2)]
   end
 
   it 'should not cache object options' do
     class Model
       attr_accessor :id, :name
-      @@objects = []
+
+      def self.objects
+        @objects ||= []
+      end
+
       def self.all
-        @@objects
+        objects
       end
-      def self.objects=(objects)
-        @@objects = objects
+
+      class << self
+        attr_writer :objects
       end
     end
 
-    s = ActiveSetting::Setting.new(:object_options => 'Model.all id name')
-    s.options.should == []
-    first = double(:id => 1, :name => 'First')
+    s = ActiveSetting::Setting.new(object_options: 'Model.all id name')
+    s.options.should eq []
+    first = double(id: 1, name: 'First')
     Model.objects = [first]
-    s.options.should == [[1, 'First']]
+    s.options.should eq [[1, 'First']]
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.